### PR TITLE
Sparky2

### PR DESCRIPTION
**Pull request by peabody124**
Adds support for the Sparky2 target. This is to start the process of code review. This will not merge until hardware design files are included.

### CODE REVIEW

File: flight/targets/sparky2/fw/Makefile [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+LDFLAGS += -lc
+LDFLAGS += $(patsubst %,-l%,$(EXTRA_LIBS))
+LDFLAGS += $(MATH_LIB)
+LDFLAGS += -lc -lgcc
+LDFLAGS += -Wl,--warn-common
+LDFLAGS += -Wl,--fatal-warnings
+
+#Linker scripts                                                                                                                                                                                                              
+LDFLAGS += -T memory.ld $(addprefix -T,$(LINKER_SCRIPTS_APP))
+
```
Comment (mlyle) [inline flight/targets/sparky2/fw/Makefile]: The bottom of this should be altered to be consistent with commit 82efedb to reduce duplication

Comment (peabody124) [inline flight/targets/sparky2/fw/Makefile]: done

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 2]
```diff
@@ -0,0 +1,898 @@
...
+/**
+ * Encode GPS UTC time
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]; 0=date, 1=time
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_gps_time(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Maybe we should actually implement ?

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 3]
```diff
@@ -0,0 +1,898 @@
...
+/**
+ * Encode true airspeed(TAS)
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_airspeed(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: FIXME: Implement or delete

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 4]
```diff
@@ -0,0 +1,898 @@
...
+/**
+ * Encode gps coordinates
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]; 0=lattitude, 1=longitude
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_gps_coord(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Fixme: implement or delete

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 5]
```diff
@@ -0,0 +1,898 @@
...
+/**
+ * Encode accelerometer values
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]; 0=X, 1=Y, 2=Z
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_acc(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Fixme: Implement or delete

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 6]
```diff
@@ -0,0 +1,898 @@
...
+ * 3 - 3D fix
+ * 4 - 3D fix and HomeLocation is SET - should be safe for navigation
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_t2(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Fixme: Implement or delete

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 7]
```diff
@@ -0,0 +1,898 @@
...
+
+/**
+ * Encode heading value
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_gps_course(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Fixme: implement or delete

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 8]
```diff
@@ -0,0 +1,898 @@
...
+/**
+ * Encode gps altitude
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_gps_alt(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*if (GPSPositionHandle() == NULL)
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Fixme: Implement or delete

File: flight/Modules/UAVOTaranis/UAVOTaranis.c [hunk 9]
```diff
@@ -0,0 +1,898 @@
...
+/**
+ * Encode gps speed
+ * @param[out] value encoded value
+ * @param[in] test_presence_only true when function should only test for availability of this value
+ * @param[in] arg argument specified in frsky_value_items[]
+ * @returns true when value succesfully encoded or presence test passed
+ */
+static bool frsky_encode_gps_speed(uint32_t *value, bool test_presence_only, uint32_t arg)
+{
+	/*
```
Comment (elafargue) [inline flight/Modules/UAVOTaranis/UAVOTaranis.c]: Fixme: Implement or delete

File: flight/targets/pipxtreme/board-info/board-info.mk [hunk 10]
```diff
@@ -1,5 +1,5 @@
 BOARD_TYPE          := 0x03
```
Comment (elafargue) [inline flight/targets/pipxtreme/board-info/board-info.mk]: -> This is TauLink, right ? Should probably be in its own PR

File: flight/PiOS/Common/pios_mpu9250_spi.c [hunk 11]
```diff
@@ -539,7 +539,9 @@ void PIOS_MPU9250_SetGyroLPF(enum pios_mpu9250_gyro_filter filter)
  */
 void PIOS_MPU9250_SetAccelLPF(enum pios_mpu9250_accel_filter filter)
 {
-	PIOS_MPU9250_WriteReg(PIOS_MPU9250_ACCEL_DLPF_CFG_REG, filter);
```
Comment (elafargue) [inline flight/PiOS/Common/pios_mpu9250_spi.c]: Confused here - 9250 or 60X0 ???

**Comment by elafargue** [comment]
LGTM ! Looking forward to merging this one

File: flight/PiOS/Common/pios_mpu9250_spi.c (hunk 11 above)
Comment (peabody124) [inline flight/PiOS/Common/pios_mpu9250_spi.c]: All these constants come from mpu60x0 except the mag, because that component of the chip did not change. i was making this constant match what is in the datasheet.
